src/methods/clipping/clipping_processor.jl

```diff
@@ -0,0 +1,295 @@
+# # This file contains the shared helper functions for the other polygon clipping
+# # functionalities.
+
+"""
+    _poly_in_poly(::GI.PolygonTrait, poly_a, ::GI.PolygonTrait, poly_b):: Bool 
+
+    This function is only called when we know the two polygons do not intersect. This
+    returns true if poly_a is completely contained in poly_b and false otherwise.
+"""
+# TODO: tempory _poly_in_poly, only works if entirely contained
+# Checks if polya in poly b
+function _poly_in_poly(poly_a, poly_b)
```

Comment:
Do we need this helper function? Can we just do the point in polygon check where it is used? Especially because the name is misleading, although you do cover that in the comments